src/controllers/project.js

```diff
@@ -0,0 +1,21 @@
+const Project = require('../models/project');
+
+const filterProjects = async (req, res) => {
+  try {
+    const { q } = req.query;
```

Comment:
here q will be written in the url when searching so it should more readable name like category maybe

Reply:
in the search bar on the site, the user will write anything 
when the user wants to filter the projects by only category the URL can be 
`somename.com?q={category-name}`
so the query name MOST be q that is the standard 

https://www.trendyol.com/sr?**q=asdsdsdf**&qt=asdsdsdf&st=asdsdsdf&os=1

this from trendyol they are passing q as their query parameter name

https://laracasts.com/search?**query=sdfsdf**

this one is from laracasts the search query name is query (q for short)
if the user wanted detailed filtering we have to change the endpoint to make all the cases we have
but since only category, maybe title and description keep it simple there is no need to make it complicated

Reply:
thanks for letting me know, but i think we don't have only category to filter we have a search filter also by text which will be implemented later, if you feel that this makes sense then go for it